### PR TITLE
One Copyright Update to rule them all

### PR DESCRIPTION
**Pull request by akien-mga**
**TL;DR** as the diff is huge:

This copyright notice:
```
Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.
Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).
```
changes to:
```
Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md).
Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.
```

---

As many open source projects have started doing it, we're removing the current year from the copyright notice, so that we don't need to bump it every year.

It seems like only the first year of publication is technically relevant for copyright notices, and even that seems to be something that many companies stopped listing altogether (in a version controlled codebase, the commits are a much better source of date of publication than a hardcoded copyright statement).

We also now list Godot Engine contributors first as we're collectively the current maintainers of the project, and we clarify that the "exclusive" copyright of the co-founders covers the timespan before opensourcing (their further contributions are included as part of Godot Engine contributors).

Also fixed "cf." Frenchism - it's meant as "refer to / see".

---

Inspired by https://hynek.me/til/copyright-years/ which links to many examples of open source projects from big tech companies which either adopted the `-present` format we picked, or removed the end year, or removed all years altogether.

We discussed which format to use with @godotengine/production and agreed on using the `-present` format for the currently "active" copyright notice, and keeping the 2007-2014 range for @reduz and @punto-'s original open sourcing copyright notice.

### CODE REVIEW

**Review by clayjohn — APPROVED** [review]

**Review by reduz — APPROVED** [review]
You definitely earned the Hacktoberfest t-shirt with this one.

**Comment by bitbrain** [comment]
Amazing!
